### PR TITLE
mithril: Apply more lint rules

### PR DESCRIPTION
**Pull request by ghost**
Most of these were easy to apply. `no-empty-interface` will require a bit more work.

### CODE REVIEW

File: types/mithril/index.d.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ declare namespace Mithril {
 		onupdate?(this: State, vnode: VnodeDOM<Attrs, State>): any;
 	}
 
-	export interface Hyperscript {
+	interface Hyperscript {
```
Comment (isiahmeadows) [inline types/mithril/index.d.ts]: Are interfaces and type aliases within namespaces auto-exported?

**Review by dead-claudia — APPROVED** [review]
LGTM mod my one non-blocking question.

**Comment by spacejack** [comment]
Works okay for me. Also did not know interfaces auto export from namespaces. Sorry about the inconsistent styles in the tests.

**Review by spacejack — APPROVED** [review]

**Comment by isiahmeadows** [comment]
@andy-ms Are you aware this has merge conflicts?

File: types/mithril/index.d.ts (hunk 1 above)
Comment (andy-ms) [inline types/mithril/index.d.ts]: Yep.

**Pull request merged**